### PR TITLE
fixes to lsf_monitoring and aux preparator

### PR DESCRIPTION
**Pull request by dougbenjamin**

### CODE REVIEW

File: pandaharvester/harvesterpreparator/analysis_aux_preparator.py [hunk 1]
```diff
@@ -50,8 +50,14 @@ def trigger_preparation(self, jobspec):
                     tmpLog.debug('getting via http from {0} to {1}'.format(url, accPath))
                     res = requests.get(url, timeout=180, verify=False)
                     if res.status_code == 200:
-                        with open(accPath, 'w') as f:
-                            f.write(res.content)
+                        tmpLog.debug('res.headers = {0}'.format(res.headers))
+                        # Should we check the returned size if too small through error?
+                        if res.headers['content-type'] == 'application/gzip' :
```
Comment (tmaeno) [inline pandaharvester/harvesterpreparator/analysis_aux_preparator.py]: Isn't it enough to open the file in the binary mode? Do you need this if-else?

Comment (dougbenjamin) [reply pandaharvester/harvesterpreparator/analysis_aux_preparator.py]: binary mode works with runcontainer transform

Comment (dougbenjamin) [reply pandaharvester/harvesterpreparator/analysis_aux_preparator.py]: removed the if else logic in latest commit.

**Comment by dougbenjamin** [comment]
removed the if else for writing the file from http and added protection for 0 length file. remove 0 length files.

**Comment by tmaeno** [comment]
Empty files can happen with other protocols as well. Isn't it better to fetch files to tmp files first
and then rename them to accPath when return_code=0 at around https://github.com/HSF/harvester/blob/7d5aa29467c6875bb727194d51fb1c098ef69ba5/pandaharvester/harvesterpreparator/analysis_aux_preparator.py#L114

rather than adding the following code snippet
```
                if os.path.getsize(accPath) == 0 :
                    tmpLog.debug('Remove file - {0}'.format(accPath))
                    os.remove(accPath)
```
here and there?

Also 

> test if we are at Summit - container needs to be built elsewhere

why you don't simply install singularity on DTN at summit?

File: pandaharvester/harvesterpreparator/analysis_aux_preparator.py [hunk 2]
```diff
@@ -50,31 +57,52 @@ def trigger_preparation(self, jobspec):
...
-                    if stderr is not None:
-                        stderr = stderr.replace('\n', ' ')
-                    tmpLog.debug("stdout: %s" % stdout)
-                    tmpLog.debug("stderr: %s" % stderr)
-                except Exception:
-                    core_utils.dump_error_message(tmpLog)
+                # test if docker command exists
+                # else test if singularity command exists
+                args = None
+                if shutil.which('docker') is not None:
```
Comment (tmaeno) [inline pandaharvester/harvesterpreparator/analysis_aux_preparator.py]: BTW this logic is not good to use singularity when both docker and singularity are available. I suggest adding a plugin config parameter like containerRuntime and have an if-else using the parameter.

**Comment by tmaeno** [comment]
Closing since a new pull request was made in https://github.com/HSF/harvester/pull/38 based on this.

**Pull request closed without merging**